addons/osio-addon/src/main/java/io/fabric8/launcher/osio/steps/booster/GolangBooster.java

```diff
@@ -0,0 +1,264 @@
+import org.codehaus.plexus.util.FileUtils;
+
+import io.fabric8.launcher.core.api.Projectile;
+import io.fabric8.launcher.core.spi.ProjectileEnricher;
+import io.fabric8.launcher.osio.projectiles.OsioLaunchProjectile;
+import io.fabric8.launcher.service.git.api.GitService;
+
+/**
+ * Customizes golan booster files.
+ *
+ */
+@ApplicationScoped
+public class GolangBooster implements ProjectileEnricher {
+
+    @Inject
+    GitService gitService;
+
+    private static final Logger log = Logger.getLogger(GolangBooster.class.getName());
+    private static final String GO_FILE_EXTENSION = ".go";
+    private static final String ENVIRONMENT = "environment";
+    private static final String ASSEMBLE = "assemble";
+    private static final String MAKEFILE = "Makefile";
+    private static final String GIT = "git";
+    private static final String URL = "url";
+    private static final String SOURCE = "source";
+    private static final String START_IMPORT = "import (";
+    private static final String END_IMPORT = ")";
+    private static final String NEW_LINE = "\n";
+    private static final String NO_FILE_EXTENSION = "";
+    private static final String SEPARATOR = "/";
+
+    private Path projectLocation;
+    private Map<String, Object> boosterData;
+    private String gitUser;
+    private Map<File, String> filesToPush = new HashMap<>();
+    private String projectName;
+    private String osioProjectName;
+    private String gitOrg;
+
+    @Override
+    public void accept(Projectile arg0) {
+        OsioLaunchProjectile proj = (OsioLaunchProjectile) arg0;
+        this.projectLocation = proj.getProjectLocation();
+        this.boosterData =  proj.getBooster().getData();
+        this.gitUser = gitService.getLoggedUser().getLogin();
+        this.projectName = proj.getGitRepositoryName();
+        customize();
+    }
+
+    /**
+     * Gets a list of files from a directory and writes the modified file contents.
+     */
+    public void customize() {
+        try {
+            Files.walk(this.projectLocation)
+            .forEach(filePath -> {
+              customizeBoosterFiles(filePath.toFile());
+            });
+        } catch (IOException e) {
+            log.log(Level.SEVERE, "Error while walking files in repository", e);
+        }
+
+        for (Entry<File, String> entry : filesToPush.entrySet()) {
+            try {
+                Files.write(entry.getKey().toPath(), entry.getValue().getBytes());
+            } catch (IOException e) {
+                log.log(Level.SEVERE, "Error while replacing files in repository", e);
+            }
+        }
+    }
+
+    /**
+     * If the file is either a .go file or either makefile or environment file
+     * customize the file by replacing the git organization. Once customized
+     * push the file to the repository.
+     *
+     * @param file
+     *            File to template
+     */
+    private void customizeBoosterFiles(File file) {
```

Comment:
This method could return the `filesToPush` map (which would be better called `filesToFlush` or `filesToOverwrite`) instead of having it as a class attribute

Reply:
or rather, why not perform the writing directly? That would be much faster no?